src/Access/SettingsProfilesCache.cpp

```diff
@@ -142,12 +141,22 @@ void SettingsProfilesCache::mergeSettingsAndConstraintsFor(EnabledSettings & ena
     merged_settings.merge(enabled.params.settings_from_enabled_roles);
     merged_settings.merge(enabled.params.settings_from_user);
 
+    std::vector<UUID> applied_profiles;
+    applied_profiles.reserve(merged_settings.size());
+    for (const auto & setting_element : merged_settings)
+    {
+        if (setting_element.parent_profile)
```

Comment:
It's a bit more complicated. A parent profile can have its own parent profile, and so on. `substituteProfiles()` is a better place to collect all such parent profiles. And are you sure we need those parent profiles in `system.session_log`? Just a list of enabled roles isn't enough?

Reply:
IIRC profiles can be assigned not only to roles, but to users too, and maybe to something else. So roles alone are definitely not enough.